packages/beacon-node/src/api/impl/beacon/state/index.ts

```diff
@@ -63,7 +63,7 @@ export function getBeaconStateApi({
       if (filters?.id) {
         for (const id of filters.id) {
           const validatorIndex = getStateValidatorIndex(id, state, pubkey2index);
-          if (validatorIndex != null) {
+          if (validatorIndex !== undefined) {
```

Comment:
`getStateValidatorIndex` returns `number | undefined`. This ensures that `null` can never be returned. If `null` is attempted, this will fail at compile time. 

The change catches a bug, because if `getStateValidatorIndex` returns `undefined`, then `validatorIndex != null` will return true. Which is not what is intended